blog/test.py

```diff
@@ -11,72 +11,50 @@
 from trips import views
 
 
-class ArticleTest(TestCase):
-    def setUp(self):
-        self.client = app.test_client()
-
-    def tearDown(self):
-        User.objects.delete()
-        Article.objects.delete()
-
-    @staticmethod
-    def create_user():
-        user = User(email='emailtest@test.com',
-                    username='emailtest').set_password('testtest').save()
-        return user
-
-    def login_user(self):
-        data = {
-            'email': 'emailtest@test.com',
-            'password': 'testtest',
-        }
-        self.client.post('/login', data=data, follow_redirects=True)
-
+class ArticleTest(BaseTest):
     def test_articles_empty(self):
         result = self.client.get('/articles/')
         self.assertEqual(result.status_code, 200)
 
     def test_articles_with_an_article(self):
-        article = Article(title='<h1>title</h1>', content='<p>content</p>')
-        user = self.create_user()
-        article.author = user
-        article.save()
+        user = self.user
+        Article(title='Title for an article',
+                content='<p>content</p>',
+                author=user).save()
         result = self.client.get('/articles/')
-        self.assertIn(b'<article class=preview>', result.data)
+        self.assertIn(b'Title for an article', result.data)
 
     def test_read_article_unauthenticated(self):
-        article = Article(title='title',
+        article = Article(title='title for an article',
                           content='<p>content</p>',
-                          author=self.user()).save()
+                          author=self.user).save()
         result = self.client.get('/@emailtest/{}-{}/'.format(article.slug,
-                                                            str(article.id)))
+                                                             str(article.id)))
+        self.assertIn(b'title for an article', result.data)
```

Comment:
check the status first, then the body.
The purpose of doing so is that if the status is wrong, the error is more explicit